crates/storage/src/vm_storage.rs

```diff
@@ -367,12 +369,15 @@ where
 
         // verify key is in range
         current_key
-            .checked_add(U256::from(values.len()))
+            .checked_add(U256::from(values.len().saturating_sub(1)))
             .ok_or_else(|| anyhow!("range op exceeded available keyspace"))?;
 
         let mut key_bytes = Bytes32::zeroed();
         let mut found_unset = 0u32;
-        for value in values {
+        for (idx, value) in values.iter().enumerate() {
+            if idx != 0 {
+                current_key.increase()?;
+            }
             current_key.to_big_endian(key_bytes.as_mut());
```

Comment:
```suggestion
if let Some((first_idx, first_value)) = values.iter().enumerate().next() {
    current_key.to_big_endian(key_bytes.as_mut());

    for (idx, value) in values.iter().enumerate().skip(1) {
        current_key.increase()?; 
        current_key.to_big_endian(key_bytes.as_mut());
    }
}

```

Reply:
If I try to make it compile and remove the unused part we endup with : 
``` rust
if let Some(value) = values.iter().next() {
            current_key.to_big_endian(key_bytes.as_mut());
            let option = self
                .database
                .storage::<ContractsState>()
                .replace(&(contract_id, &key_bytes).into(), value)?;

            if option.is_none() {
                found_unset = found_unset
                    .checked_add(1)
                    .expect("We've checked it above via `values.len()`");
            }
            for value in values.iter().skip(1) {
                current_key.increase()?; 
                current_key.to_big_endian(key_bytes.as_mut());
                let option = self
                .database
                .storage::<ContractsState>()
                .replace(&(contract_id, &key_bytes).into(), value)?;

                if option.is_none() {
                    found_unset = found_unset
                        .checked_add(1)
                        .expect("We've checked it above via `values.len()`");
                }
            }
        }
```
Which seems a lot of code duplication to me, wdyt ?

Reply:
eh yeah that looks not so nice, approving!